hbase-hadoop-compat/src/test/java/org/apache/hadoop/metrics2/impl/TestMutableRangeHistogram.java

```diff
@@ -0,0 +1,66 @@
+package org.apache.hadoop.metrics2.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import org.apache.hadoop.hbase.HBaseClassTestRule;
+import org.apache.hadoop.hbase.testclassification.MetricsTests;
+import org.apache.hadoop.hbase.testclassification.SmallTests;
+import org.apache.hadoop.metrics2.AbstractMetric;
+import org.apache.hadoop.metrics2.MetricsRecordBuilder;
+import org.apache.hadoop.metrics2.lib.MutableRangeHistogram;
+import org.apache.hadoop.metrics2.lib.MutableSizeHistogram;
+import org.apache.hadoop.metrics2.lib.MutableTimeHistogram;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Collection;
+
+@Category({ MetricsTests.class, SmallTests.class})
+public class TestMutableRangeHistogram {
+
+    @ClassRule
+    public static final HBaseClassTestRule CLASS_RULE =
+            HBaseClassTestRule.forClass(TestMutableRangeHistogram.class);
+
+    private static final String RECORD_NAME = "test";
+    private static final String SIZE_METRIC_NAME = "TestSize";
+    private static final String TIME_METRIC_NAME = "TestTime";
+
+    @Test
+    public void testMutableSizeHistogram() {
+        testMutableRangeHistogram(SIZE_METRIC_NAME, new MutableSizeHistogram(SIZE_METRIC_NAME, ""));
+    }
+
+    @Test
+    public void testMutableTimeHistogram() {
+        testMutableRangeHistogram(TIME_METRIC_NAME, new MutableTimeHistogram(TIME_METRIC_NAME, ""));
+    }
+
+    private static void testMutableRangeHistogram(String metricName, MutableRangeHistogram histogram) {
+        // fill up values
+        long[] ranges = histogram.getRanges();
+        for (long val : ranges) {
+            histogram.add(val - 1);
+        }
+        histogram.add(ranges[ranges.length - 1] + 1);
```

Comment:
an attempt to choose a value well above the max to ensure it ends up in the highest bin.
```suggestion
        histogram.add(ranges[ranges.length - 1] + ranges[ranges.length - 2] / 2);
```